sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/BeamZetaSqlCalcRel.java

```diff
@@ -190,10 +193,22 @@ private static TimestampedFuture create(Instant t, Future<Value> f) {
       this.outputSchema = outputSchema;
       this.defaultTimezone = defaultTimezone;
       this.verifyRowValues = verifyRowValues;
+
+      PreparedExpression exp = prepareExpression(sql, nullParams, inputSchema, defaultTimezone);
+
+      ImmutableList.Builder<Integer> columns = new ImmutableList.Builder<>();
+      for (String c : exp.getReferencedColumns()) {
+        columns.add(Integer.parseInt(c.substring(1)));
+      }
+      this.referencedColumns = columns.build();
+      this.fieldAccess = FieldAccessDescriptor.withFieldIds(this.referencedColumns);
+
+      exp.close();
```

Comment:
PreparedExpression implements AutoCloseable, so we can use a try-with-resources block instead.

Reply:
Done.